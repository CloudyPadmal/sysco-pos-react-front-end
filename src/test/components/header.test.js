import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { shallowWrap, mountWrap, mount } from "../enzyme";
import { BrowserRouter as Router } from "react-router-dom";

import ConnectedHeader, { Header, ConditionalHeader } from "../../components/Header";

const middlewares = [thunk];

describe("<Header /> component", () => {
    it("renders the component", () => {
        const wrapper = shallowWrap(<Header isLoggedIn={true} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("mounts the component with login true", () => {
        const wrapper = mountWrap(<Header isLoggedIn={true} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("mounts the component with login false", () => {
        const wrapper = mountWrap(<Header isLoggedIn={false} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<ConditionalHeader /> component", () => {
    it("renders the component", () => {
        const wrapper = shallowWrap(<ConditionalHeader isLoggedIn={true} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("mounts the component with login true", () => {
        const wrapper = mountWrap(<ConditionalHeader isLoggedIn={true} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("mounts the component with login false", () => {
        const wrapper = mountWrap(<ConditionalHeader isLoggedIn={false} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<Header /> component with logged out", () => {

    const initialState = {
        uac: { isLoggedIn: false }
    };
    const mockStore = configureStore(middlewares);
    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    });

    it("renders with login and register links", () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router>
                    <ConnectedHeader />
                </Router>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<Header /> component with logged in", () => {

    const initialState = {
        uac: { isLoggedIn: true }
    };
    const mockStore = configureStore(middlewares);
    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    });

    it("renders with order and logout links", () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router>
                    <ConnectedHeader />
                </Router>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});