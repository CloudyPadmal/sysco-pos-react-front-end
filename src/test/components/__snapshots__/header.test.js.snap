// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConditionalHeader /> component mounts the component with login false 1`] = `
<ConditionalHeader
  isLoggedIn={false}
>
  <div
    className="collapse navbar-collapse"
  >
    <Link
      className="navbar-brand"
      replace={false}
      style={
        Object {
          "paddingTop": "0px",
        }
      }
      to="/"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "paddingTop": "0px",
          }
        }
      >
        Sysco PoS System
      </a>
    </Link>
    <div
      className="navbar-nav ml-auto"
    >
      <Link
        className="nav-item nav-link"
        replace={false}
        to="/login"
      >
        <a
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-in-alt"
          />
           Login
        </a>
      </Link>
      <Link
        className="nav-item nav-link"
        replace={false}
        to="/register"
      >
        <a
          className="nav-item nav-link"
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </Link>
    </div>
  </div>
</ConditionalHeader>
`;

exports[`<ConditionalHeader /> component mounts the component with login true 1`] = `
<ConditionalHeader
  isLoggedIn={true}
>
  <div
    className="collapse navbar-collapse"
  >
    <Link
      className="navbar-brand"
      replace={false}
      style={
        Object {
          "paddingTop": "0px",
        }
      }
      to="/"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "paddingTop": "0px",
          }
        }
      >
        Sysco PoS System
      </a>
    </Link>
    <div
      className="navbar-nav mr-auto mt-2 mt-lg-0"
    >
      <Link
        className="nav-item nav-link"
        replace={false}
        to="/my_orders"
      >
        <a
          className="nav-item nav-link"
          href="/my_orders"
          onClick={[Function]}
        >
          <i
            className="fas fa-list-ul"
          />
           My Orders
        </a>
      </Link>
      <Link
        className="nav-item nav-link"
        replace={false}
        to="/create_order"
      >
        <a
          className="nav-item nav-link"
          href="/create_order"
          onClick={[Function]}
        >
          <i
            className="fas fa-plus"
          />
           Create Order
        </a>
      </Link>
    </div>
    <div
      className="navbar-nav my-2 my-lg-0"
    >
      <Link
        className="nav-item nav-link"
        replace={false}
        to="/logout"
      >
        <a
          className="nav-item nav-link"
          href="/logout"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-out-alt"
          />
           Logout
        </a>
      </Link>
    </div>
  </div>
</ConditionalHeader>
`;

exports[`<ConditionalHeader /> component renders the component 1`] = `
<div
  className="collapse navbar-collapse"
>
  <Link
    className="navbar-brand"
    replace={false}
    style={
      Object {
        "paddingTop": "0px",
      }
    }
    to="/"
  >
    Sysco PoS System
  </Link>
  <div
    className="navbar-nav mr-auto mt-2 mt-lg-0"
  >
    <Link
      className="nav-item nav-link"
      replace={false}
      to="/my_orders"
    >
      <i
        className="fas fa-list-ul"
      />
       My Orders
    </Link>
    <Link
      className="nav-item nav-link"
      replace={false}
      to="/create_order"
    >
      <i
        className="fas fa-plus"
      />
       Create Order
    </Link>
  </div>
  <div
    className="navbar-nav my-2 my-lg-0"
  >
    <Link
      className="nav-item nav-link"
      replace={false}
      to="/logout"
    >
      <i
        className="fas fa-sign-out-alt"
      />
       Logout
    </Link>
  </div>
</div>
`;

exports[`<Header /> component mounts the component with login false 1`] = `
<Header
  isLoggedIn={false}
>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <ConditionalHeader
        isLoggedIn={false}
      >
        <div
          className="collapse navbar-collapse"
        >
          <Link
            className="navbar-brand"
            replace={false}
            style={
              Object {
                "paddingTop": "0px",
              }
            }
            to="/"
          >
            <a
              className="navbar-brand"
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "paddingTop": "0px",
                }
              }
            >
              Sysco PoS System
            </a>
          </Link>
          <div
            className="navbar-nav ml-auto"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/login"
            >
              <a
                className="nav-item nav-link"
                href="/login"
                onClick={[Function]}
              >
                <i
                  className="fas fa-sign-in-alt"
                />
                 Login
              </a>
            </Link>
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/register"
            >
              <a
                className="nav-item nav-link"
                href="/register"
                onClick={[Function]}
              >
                Register
              </a>
            </Link>
          </div>
        </div>
      </ConditionalHeader>
    </nav>
  </header>
</Header>
`;

exports[`<Header /> component mounts the component with login true 1`] = `
<Header
  isLoggedIn={true}
>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <ConditionalHeader
        isLoggedIn={true}
      >
        <div
          className="collapse navbar-collapse"
        >
          <Link
            className="navbar-brand"
            replace={false}
            style={
              Object {
                "paddingTop": "0px",
              }
            }
            to="/"
          >
            <a
              className="navbar-brand"
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "paddingTop": "0px",
                }
              }
            >
              Sysco PoS System
            </a>
          </Link>
          <div
            className="navbar-nav mr-auto mt-2 mt-lg-0"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/my_orders"
            >
              <a
                className="nav-item nav-link"
                href="/my_orders"
                onClick={[Function]}
              >
                <i
                  className="fas fa-list-ul"
                />
                 My Orders
              </a>
            </Link>
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/create_order"
            >
              <a
                className="nav-item nav-link"
                href="/create_order"
                onClick={[Function]}
              >
                <i
                  className="fas fa-plus"
                />
                 Create Order
              </a>
            </Link>
          </div>
          <div
            className="navbar-nav my-2 my-lg-0"
          >
            <Link
              className="nav-item nav-link"
              replace={false}
              to="/logout"
            >
              <a
                className="nav-item nav-link"
                href="/logout"
                onClick={[Function]}
              >
                <i
                  className="fas fa-sign-out-alt"
                />
                 Logout
              </a>
            </Link>
          </div>
        </div>
      </ConditionalHeader>
    </nav>
  </header>
</Header>
`;

exports[`<Header /> component renders the component 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <ConditionalHeader
      isLoggedIn={true}
    />
  </nav>
</header>
`;

exports[`<Header /> component with logged in renders with order and logout links 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          dispatch={[Function]}
          isLoggedIn={true}
        >
          <header>
            <nav
              className="navbar navbar-expand-lg navbar-light bg-light"
            >
              <ConditionalHeader
                isLoggedIn={true}
              >
                <div
                  className="collapse navbar-collapse"
                >
                  <Link
                    className="navbar-brand"
                    replace={false}
                    style={
                      Object {
                        "paddingTop": "0px",
                      }
                    }
                    to="/"
                  >
                    <a
                      className="navbar-brand"
                      href="/"
                      onClick={[Function]}
                      style={
                        Object {
                          "paddingTop": "0px",
                        }
                      }
                    >
                      Sysco PoS System
                    </a>
                  </Link>
                  <div
                    className="navbar-nav mr-auto mt-2 mt-lg-0"
                  >
                    <Link
                      className="nav-item nav-link"
                      replace={false}
                      to="/my_orders"
                    >
                      <a
                        className="nav-item nav-link"
                        href="/my_orders"
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-list-ul"
                        />
                         My Orders
                      </a>
                    </Link>
                    <Link
                      className="nav-item nav-link"
                      replace={false}
                      to="/create_order"
                    >
                      <a
                        className="nav-item nav-link"
                        href="/create_order"
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-plus"
                        />
                         Create Order
                      </a>
                    </Link>
                  </div>
                  <div
                    className="navbar-nav my-2 my-lg-0"
                  >
                    <Link
                      className="nav-item nav-link"
                      replace={false}
                      to="/logout"
                    >
                      <a
                        className="nav-item nav-link"
                        href="/logout"
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-sign-out-alt"
                        />
                         Logout
                      </a>
                    </Link>
                  </div>
                </div>
              </ConditionalHeader>
            </nav>
          </header>
        </Header>
      </Connect(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<Header /> component with logged out renders with login and register links 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          dispatch={[Function]}
          isLoggedIn={false}
        >
          <header>
            <nav
              className="navbar navbar-expand-lg navbar-light bg-light"
            >
              <ConditionalHeader
                isLoggedIn={false}
              >
                <div
                  className="collapse navbar-collapse"
                >
                  <Link
                    className="navbar-brand"
                    replace={false}
                    style={
                      Object {
                        "paddingTop": "0px",
                      }
                    }
                    to="/"
                  >
                    <a
                      className="navbar-brand"
                      href="/"
                      onClick={[Function]}
                      style={
                        Object {
                          "paddingTop": "0px",
                        }
                      }
                    >
                      Sysco PoS System
                    </a>
                  </Link>
                  <div
                    className="navbar-nav ml-auto"
                  >
                    <Link
                      className="nav-item nav-link"
                      replace={false}
                      to="/login"
                    >
                      <a
                        className="nav-item nav-link"
                        href="/login"
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-sign-in-alt"
                        />
                         Login
                      </a>
                    </Link>
                    <Link
                      className="nav-item nav-link"
                      replace={false}
                      to="/register"
                    >
                      <a
                        className="nav-item nav-link"
                        href="/register"
                        onClick={[Function]}
                      >
                        Register
                      </a>
                    </Link>
                  </div>
                </div>
              </ConditionalHeader>
            </nav>
          </header>
        </Header>
      </Connect(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;
